src/viewer.js

```diff
@@ -3293,6 +3295,68 @@ function onCanvasPinch( event ) {
     }
 }
 
+function onCanvasFocus(event) {
+    /**
+     * Raised when a pointer enters the {@link OpenSeadragon.Viewer#canvas} element.
+     *
+     * @event canvas-enter
+     * @memberof OpenSeadragon.Viewer
+     * @type {object}
+     * @property {OpenSeadragon.Viewer} eventSource - A reference to the Viewer which raised this event.
+     * @property {OpenSeadragon.MouseTracker} tracker - A reference to the MouseTracker which originated this event.
+     * @property {String} pointerType - "mouse", "touch", "pen", etc.
```

Comment:
Most of these properties aren't present in the focus/blur events. The only properties you'll want to document here (and for the blur event) are `eventSource`, `tracker`, `originalEvent`, and `userData`.